pkg/api/cached.go

```diff
@@ -0,0 +1,99 @@
+package api
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/gadget-inc/dateilager/internal/files"
+	"github.com/gadget-inc/dateilager/internal/key"
+	"github.com/gadget-inc/dateilager/internal/logger"
+	"github.com/gadget-inc/dateilager/internal/pb"
+	"github.com/gadget-inc/dateilager/pkg/client"
+	"golang.org/x/sys/unix"
+)
+
+type Cached struct {
+	pb.UnimplementedCacheServer
+	Client      *client.Client
+	StagingPath string
+	// the current directory holding a fully formed downloaded cache
+	currentDir string
+	// the current version of the cache on disk at currentDir
+	currentVersion int64
+}
+
+func (c *Cached) PopulateDiskCache(ctx context.Context, req *pb.PopulateDiskCacheRequest) (*pb.PopulateDiskCacheResponse, error) {
+	err := requireAdminAuth(ctx)
```

Comment:
Shouldn't this require shared reader?

Reply:
Because this actually modifies the filesystem at an arbitrary path using the permissions of the daemon, and because we don't actually want to use this for much other than testing locally, I figured better to make it restrictive and accept admin only. I think if we ever have a non-CSI use case for this we can lower it? Or, I can just delete this whole piece too

Reply:
We use this pattern for dev only routes (especially for things like resetting a DB) and it's probably a good call here: https://github.com/gadget-inc/dateilager/blob/main/pkg/api/fs.go#L783-L785